src/users/UserItem.js

```diff
@@ -57,7 +57,8 @@ export class UserItemRaw<
   handlePress = () => {
     const { user, onPress } = this.props;
     // TODO cut this `user.email` condition -- it should never trigger, and
-    //   looks like a fudge for the possibility of data coming from NULL_USER
+    //   looks like a fudge for the possibility of data coming from
+    //   the late NULL_USER
```

Comment:
> It also seems obnoxious for `UserItem` to sometimes ignore the `onPress` that a caller has decided to pass—especially since the caller _also_ controls the `user` that this condition is based on

Yeah, agreed.



> I think the `onPress` condition (i.e., checking if `onPress` exists) can certainly be removed. It's marked as required, and all callers pass it.

Hmm, true. I think the right fix for that is actually in the other direction, though: the `onPress` should be marked as optional. At least one caller (in `ShareToPm`) passes a no-op callback for it; even if that weren't the case now, there could be callers in the future that just wanted to display the user and not offer an interaction related to them.

Making the distinction between having a real action to do here, and not having one, is good because among other things it can let us pass that distinction on to the underlying UI widgets, and that can give an important hint to a11y tools that some widgets don't have an interaction and shouldn't be offered to try to interact with. (I think this kind of thing is the cause of at least one of the items listed at #4394: "On the PM-conversations screen, the avatars are selectable. They shouldn't be".)

Reply:
> Hmm, true. I think the right fix for that is actually in the other direction, though

Ah, yeah, those are good points.